x/ccv/provider/keeper/relay.go

```diff
@@ -42,6 +41,21 @@ func (k Keeper) OnRecvVSCMaturedPacket(
 		panic(fmt.Errorf("VSCMaturedPacket received on unknown channel %s", packet.DestinationChannel))
 	}
 
+	// If no packets are in the per chain queue, immediately handle the vsc matured packet data
+	if k.GetPendingPacketDataSize(ctx, chainID) == 0 {
+		k.HandleVSCMaturedPacket(ctx, chainID, data)
+	} else {
+		// Otherwise queue the packet data as pending (behind one or more pending slash packet data instances)
+		k.QueuePendingVSCMaturedPacketData(ctx, chainID, packet.Sequence, data)
+	}
+
+	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
+	return ack
+}
+
+func (k Keeper) HandleVSCMaturedPacket(
```

Comment:
It's still unclear to me what the desired behavior is when `HandleVSCMaturedPacket` fails. Are the panic failures somethings we can just silently drop? Or is it important enough to panic. In main, we panic

Reply:
Currently we panic inside `HandleVSCMaturedPacket` if there are some problems with marshaling data to the store. In this case it is okay to panic. We should probably panic directly in the setters / getters. However, since we do return an error, IMO this error should be handled either in `EndBlock` or on `DeliverTx` (i.e., an IBC handler). This will make it clear what the behavior is: either the system shuts down or the transaction is aborted.